### PR TITLE
Add diffoscope

### LINKED ISSUE #2247: diffoscope
### Please confirm there isn't an existing package request

- [X] I have searched through package requests

### Homepage

https://diffoscope.org/

### Maintainer

- [ ] Would you like to be the maintainer for this package?

### Why should this be included in the repository?

Useful for comparing two complicated artefacts e.g. .eopkg files

### Are we allowed to redistribute it?

GPL-3.0-or-later

### What kind of user will use this package, and how many users do you think will use this package?

misc

### Link to source archive file

https://salsa.debian.org/reproducible-builds/diffoscope/-/archive/264/diffoscope-264.tar.bz2

### PR DESCRIPTION
**Pull request by malfisya**
**Summary**

- Add diffoscope at 264 (Resolves #2247 )
- Add python-libarchive-c at 5.1

### CODE REVIEW

**Comment by ReillyBrogan** [comment]
Both of these packages need monitoring.yml files

**Comment by malfisya** [comment]
Ok now it can run basic diff, I tested it with :
```
diffoscope calc-2.13.0.1-3-1-x86_64.eopkg calc-2.15.0.4-4-1-x86_64.eopkg
````
My question is still the same : 

How many tools/rundeps should be added? [Arch style](https://gitlab.archlinux.org/archlinux/packaging/packages/diffoscope/-/blob/main/PKGBUILD?ref_type=heads), [Fedora style](https://src.fedoraproject.org/rpms/diffoscope/blob/rawhide/f/diffoscope.spec), [Alpine style](https://git.alpinelinux.org/aports/tree/community/diffoscope/APKBUILD), or our own ? 

It does output unpleasant error if there is missing dependency for the command you run. This is the error when I ran  `diffoscope calc-2.13.0.1-3-1-x86_64.eopkg calc-2.15.0.4-4-1-x86_64.eopkg` without `diffutils` installed : 
```
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/diffoscope/diff.py", line 337, in diff
    return run_diff(
           ^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/tools.py", line 138, in tool_check
    raise RequiredToolNotFound(command)
diffoscope.exc.RequiredToolNotFound

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/diffoscope/diff.py", line 334, in diff
    with FIFOFeeder(feeder1, fifo1_path) as fifo1, FIFOFeeder(
  File "/usr/lib/python3.11/site-packages/diffoscope/diff.py", line 225, in __exit__
    self.join()
  File "/usr/lib/python3.11/site-packages/diffoscope/diff.py", line 258, in join
    raise self._exception
  File "/usr/lib/python3.11/site-packages/diffoscope/diff.py", line 234, in run
    fd = os.open(self.fifo_path, os.O_WRONLY | os.O_NONBLOCK)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/diffoscope_iastzhvm_/tmpu4qcj4x9/fifo2'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/diffoscope", line 33, in <module>
    sys.exit(load_entry_point('diffoscope==264', 'console_scripts', 'diffoscope')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/main.py", line 767, in main
    sys.exit(run_diffoscope(parsed_args))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/main.py", line 718, in run_diffoscope
    difference = compare_root_paths(path1, path2)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/comparators/utils/compare.py", line 69, in compare_root_paths
    difference = compare_files(file1, file2)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/comparators/utils/compare.py", line 149, in compare_files
    return file1.compare(file2, source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/comparators/zip.py", line 267, in compare
    x = super().compare(other, source)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/comparators/zip.py", line 234, in compare
    x = super().compare(other, source)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/comparators/utils/file.py", line 532, in compare
    difference = self._compare_using_details(other, source)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/comparators/utils/file.py", line 433, in _compare_using_details
    details.extend(self.compare_details(other, source))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/comparators/zip.py", line 284, in compare_details
    differences.extend(zipinfo_differences(self, other))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/comparators/zip.py", line 127, in zipinfo_differences
    zipinfo = Difference.from_operation(Zipinfo, file.path, other.path)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/difference.py", line 267, in from_operation
    return Difference.from_operation_exc(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/difference.py", line 316, in from_operation_exc
    difference = Difference.from_feeder(
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/difference.py", line 214, in from_feeder
    unified_diff = diff(feeder1, feeder2)
                   ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/diffoscope/diff.py", line 334, in diff
    with FIFOFeeder(feeder1, fifo1_path) as fifo1, FIFOFeeder(
  File "/usr/lib/python3.11/site-packages/diffoscope/diff.py", line 225, in __exit__
    self.join()
  File "/usr/lib/python3.11/site-packages/diffoscope/diff.py", line 258, in join
    raise self._exception
  File "/usr/lib/python3.11/site-packages/diffoscope/diff.py", line 234, in run
    fd = os.open(self.fifo_path, os.O_WRONLY | os.O_NONBLOCK)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/diffoscope_iastzhvm_/tmpu4qcj4x9/fifo1'

```

**Comment by ReillyBrogan** [comment]
So the primary use case of diffoscope for us is to check `.eopkg`s (and later `.stone`s) to verify changes to solbuild/ypkg and for reproducibility tests. So I would say that it should have rundeps on whatever is required for a few common `.eopkg`, as well as common compression algorithms. For instance (more or less just picking a package at random that looks more complex than something truly trivial) you could build libinput and use that as a test.

File: packages/d/diffoscope/package.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    - python-defusedxml
+    - python-libarchive-c
+    - python-magic
+    - tar
+    - unzip
+    - xz
+    - zstd
+build      : |
+    %python3_setup
+install    : |
```
Comment (ReillyBrogan) [inline packages/d/diffoscope/package.yml:39]: There's a man page as well that you could add. `doc/diffoscope.1`

**Review by ReillyBrogan — CHANGES_REQUESTED** [review]
Other than that it looks good, and it worked in my testing.

File: packages/d/diffoscope/package.yml (hunk 1 above)
Comment (malfisya) [reply packages/d/diffoscope/package.yml:39]: it is kind of broken. it needs to be built first (`make -C doc`). it need `help2man` and `python-docutils`. I added those in builddeps, but it keeps failing said `rst2man` command not found and `help2man` bla..bla... So i removed those.

Comment (ReillyBrogan) [reply packages/d/diffoscope/package.yml:43]: I fixed it. It needed `help2man` and `python-docutils` like you thought, but it also tried to run `bin/diffoscope` from the build root and that failed with a ` No module named 'libarchive' No module named 'libarchive'` error. That was fixed by adding `python-libarchive-c` as a builddep as well.

**Review by ReillyBrogan — APPROVED** [review]

**Pull request merged**